packages/webapp/src/encryption/components/encryption-password-alert.tsx

```diff
@@ -0,0 +1,369 @@
+import { PrivateKey } from "eosjs/dist/PrivateKey";
+import { useState } from "react";
+
+import {
+    Button,
+    Container,
+    Form,
+    Modal,
+    onError,
+    Text,
+    useFormFields,
+    useUALAccount,
+} from "_app";
+import { generateEncryptionKey } from "_app/eos/secret-publisher";
+import { setEncryptionPublicKeyTransaction } from "encryption/transactions";
+
+import { UpdateEncryptionPassword, useEncryptionPassword } from "../hooks";
+import { useEffect } from "react";
+
+interface Props {
+    promptSetupEncryptionKey?: boolean;
+}
+
+export const EncryptionPasswordAlert = ({
+    promptSetupEncryptionKey,
+}: Props) => {
+    const {
+        encryptionPassword,
+        updateEncryptionPassword,
+    } = useEncryptionPassword();
+
+    if (!encryptionPassword) {
+        return null;
+    }
+
+    const promptNewKey =
+        promptSetupEncryptionKey && !encryptionPassword.publicKey;
+    if (promptNewKey) {
```

Comment:
I'm seeing this prompt even when signed out completely.

Reply:
Also, I see this for a quick moment while the page is loading (if I refresh). When this turns orange or red or whatever color it is, that flash will be quite noticeable.

So this should only display:

1. When a user is signed in,
2. when the user is an active eden member
3. we have their eden member record and see that they don't have a key set